### ISSUE TITLE
XDG_CONFIG_DIRS not respected

### ISSUE THREAD

**Issue by betaboon** [Alexays/Waybar#714]
i think this #425 introduced a regression on handling config-file-paths.

if i understand this correctly:

before #425 `/etc/xdg/waybar/config` was checked (which seems to have been a wrong implementation of XDG_CONFIG_DIRS in the first place)

according to [the specification](https://specifications.freedesktop.org/basedir-spec/0.6/ar01s03.html) the order of precendence should be:
- $XDG_CONFIG_HOME
- $XDG_CONFIG_DIRS (maintaining its order, and defaulting to `/etc/xdg` if empty

**Comment by jbeich** [comment]
> before #425 `/etc/xdg/waybar/config` was checked

Should still be true for most Linux distributions:
```
$ meson --prefix=/usr build
$ ninja -C build
$ strings _build/waybar | fgrep etc/xdg
//etc/xdg/waybar/style.css
//etc/xdg/waybar/config
```

**Comment by jbeich** [comment]
> `man 5 waybar` doesnt reflect the change introduced in #425 either. it still states that `/etc/xdg/waybar/config` is checked

This is similar to sway which hardcodes `/etc/sway/config` in manpages. Less preprocessing makes it easier to edit documentation.

**Comment by betaboon** [comment]
@jbeich yes it should still be true for every distribution that builds waybar (and sway) with the default prefix of `/`. in nixos we don't, thus forcing us to maintain patches that check `/etc/xdg/waybar` instead of `$prefix/etc/xdg/waybar` (as we currently do for sway).
to avoid this i am proposing to fully support the base-dir-spec linked to above, and thus supporting $XDG_CONFIG_DIRS in addition to $XDG_CONFIG_HOME.

**Comment by jbeich** [comment]
> in nixos we don't, thus forcing us to maintain patches that check `/etc/xdg/waybar`

Why do you need to patch what is supported Meson out of the box?
```
$ meson --prefix=/whatever --sysconfdir=/etc _build
$ ninja -C _build
$ strings _build/waybar | fgrep etc/xdg
//etc/xdg/waybar/style.css
//etc/xdg/waybar/config
```

**Comment by betaboon** [comment]
as far as i understand the `meson.build` does
```
add_project_arguments('-DSYSCONFDIR="/@0@"'.format(join_paths(prefix, sysconfdir)), language : 'cpp')
```
thus using `SYSCONFDIR "/xdg/waybar/config"` should result in `$PREFIX/$SYSCONFDIR/xdg/waybar/config`.

or am i missing something here ? I'm not that familiar with meson. so i might be completly wrong here ;)

**Comment by jbeich** [comment]
`join_paths` discards `prefix` if `sysconfdir` is an absolute path.
https://mesonbuild.com/Reference-manual.html#join_paths

**Comment by jbeich** [comment]
Also, the behavior is consistent with GNU autoconf.

**Comment by jbeich** [comment]
`/@0@` has extra `/` which I've probably copy-pasted from Wayfire that later dropped it. It has no effect other than cosmetic becase Meson doesn't support relative path in `--prefix`

**Comment by jbeich** [comment]
Sorry, not Wayfire but Sway.

**Comment by betaboon** [comment]
> `join_paths` discards `prefix` if `sysconfdir` is an absolute path.

ah. this was the missing link. thanks for clearing that up.

still the question remains if it would be preferable to respect `XDG_CONFIG_DIRS`
